File: mongo/jython/create/mongo_create.py
#! /usr/bin/jython
# -*- coding: utf-8 -*-
#
#	mongo_create.py
#					Sep/11/2015
#
# ----------------------------------------------------------------
import sys
sys.path.append ('/var/www/data_base/common/python_common')
from text_manipulate import dict_append_proc
#
import string
import java
#
from com.mongodb import MongoClient
from com.mongodb.client import MongoDatabase
from com.mongodb.client import MongoCollection

from org.bson import Document
#
# ----------------------------------------------------------------
def	data_prepare_proc ():
	dict_aa = {} 
#
	dict_aa = dict_append_proc (dict_aa,'t1161',u'さいたま',95184,'2004-5-21')
	dict_aa = dict_append_proc (dict_aa,'t1162',u'所沢',17825,'2004-7-12')
	dict_aa = dict_append_proc (dict_aa,'t1163',u'越谷',72654,'2004-11-17')
	dict_aa = dict_append_proc (dict_aa,'t1164',u'久喜',68412,'2004-9-9')
	dict_aa = dict_append_proc (dict_aa,'t1165',u'熊谷',47371,'2004-8-4')
	dict_aa = dict_append_proc (dict_aa,'t1166',u'秩父',35187,'2004-1-21')
	dict_aa = dict_append_proc (dict_aa,'t1167',u'川越',81296,'2004-7-23')
	dict_aa = dict_append_proc (dict_aa,'t1168',u'和光',24781,'2004-10-26')
	dict_aa = dict_append_proc (dict_aa,'t1169',u'新座',71823,'2004-12-15')
#
	return	dict_aa
#
# ----------------------------------------------------------------
def dict_to_mongo_proc (coll,dict_aa):
	for key in dict_aa.keys():
		unit = dict_aa[key]
		doc = Document("key",key).append ("name",unit['name']) \
		.append ("population",unit['population']) \
		.append ("date_mod",unit['date_mod'])
#
		coll.insertOne (doc)
# ----------------------------------------------------------------
print ("*** 開始 ***")
#
dict_aa = data_prepare_proc ()
#
mm = MongoClient ("localhost",27017)
db = mm.getDatabase ("city_db" )

col_name = "saitama"

coll = db.getCollection(col_name)

coll.drop ()
dict_to_mongo_proc (coll,dict_aa)

print coll.count()

print ("*** 終了 ***")
# ----------------------------------------------------------------

File: mongo/jruby/create/mongo_create.rb
#! /usr/bin/jruby
# -*- coding: utf-8 -*-
#
#	mongo_create.rb
#					Sep/11/2015
#
# --------------------------------------------------------
include Java

import com.mongodb.MongoClient
import com.mongodb.client.MongoDatabase
import com.mongodb.client.MongoCollection

import org.bson.Document

load '/var/www/data_base/common/ruby_common/text_manipulate.rb'
# --------------------------------------------------------
def mongo_create_proc (coll,key_in,name_in,population_in,date_mod_in)

doc = Document.new("key",key_in).append("name",name_in).append("population",population_in).append("date_mod",date_mod_in)

	coll.insertOne (doc)
end
# --------------------------------------------------------
def prepare_data_proc ()
	dict_aa = {}
	dict_aa=dict_append_proc(dict_aa,"t1161","さいたま",74126,"2007-8-12")
	dict_aa=dict_append_proc(dict_aa,"t1162","所沢",36819,"2007-1-27")
	dict_aa=dict_append_proc(dict_aa,"t1163","越谷",58472,"2007-2-8")
	dict_aa=dict_append_proc(dict_aa,"t1164","久喜",21965,"2007-9-21")
	dict_aa=dict_append_proc(dict_aa,"t1165","熊谷",84137,"2007-12-24")
	dict_aa=dict_append_proc(dict_aa,"t1166","秩父",27589,"2007-6-17")
	dict_aa=dict_append_proc(dict_aa,"t1167","川越",41693,"2007-4-28")
	dict_aa=dict_append_proc(dict_aa,"t1168","和光",52197,"2007-5-12")
	dict_aa=dict_append_proc(dict_aa,"t1169","新座",83526,"2007-11-21")
#
	return	dict_aa
end
# --------------------------------------------------------
puts	"*** 開始 ***"
#
dict_aa = prepare_data_proc()
#
mm = MongoClient.new("localhost",27017)
db = mm.getDatabase("city_db" )

col_name = "saitama"

coll = db.getCollection(col_name)

puts coll.count()

coll.drop()

dict_aa.each {|key, value|
	mongo_create_proc(coll,key,value['name'],value['population'],value['date_mod'])
}

puts "*** 終了 ***"

# --------------------------------------------------------
